src/Voron/PageLocator.cs

```diff
@@ -64,41 +61,33 @@ public PageLocator(LowLevelTransaction tx)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public bool TryGetReadOnlyPage(long pageNumber, out Page page)
         {
-            Debug.Assert(pageNumber != Invalid);
-
-            ref var node = ref _cache[pageNumber & CacheMask];
-
-            if (node.PageNumber == pageNumber && node.PageNumber != Invalid)
-            {
-                page = node.Page;
-                return true;
-            }
+            ulong bucket = (ulong)pageNumber & CacheMask;
+            Debug.Assert(bucket is >= 0 and < CacheSize);
```

Comment:
Bucket is `ulong`, is _has_ to be `>= 0`,  no?

Reply:
Yes. The `CacheMask` will force the bucket to be in the range of `0..1023` regardless if it is negative or positive (as a page number it is always positive).